File: standard-telegram-bot/src/main/java/com/immmus/standard/telegram/bot/settings/DefaultLongPollBotSettings.java
package com.immmus.standard.telegram.bot.settings;


import java.util.Optional;

final public class DefaultLongPollBotSettings implements LongPollBotSettings {
    private final TelegramBotSettings botSettings;

    public DefaultLongPollBotSettings(TelegramBotSettings botSettings) {
        this.botSettings = botSettings;
    }

    public DefaultLongPollBotSettings(String botUsername, String token) {
        this(new TelegramBotSettings.WithoutProxyDefaultTelegramBotSettings(botUsername, token));
    }

    public DefaultLongPollBotSettings(String botUsername, String token, ProxySettings proxySettings) {
        this(new TelegramBotSettings.WithProxyDefaultTelegramBotSettings(botUsername, token, proxySettings));
    }

    @Override
    public String botUsername() {
        return this.botSettings.botUsername();
    }

    @Override
    public String token() {
        return this.botSettings.token();
    }

    @Override
    public Optional<ProxySettings> proxySettings() {
        return this.botSettings.proxySettings();
    }
}

File: standard-bot/src/main/kotlin/com/immmus/common/telegram/bot/settings/DefaultLongPollBotSettings.kt
package com.immmus.common.telegram.bot.settings

class DefaultLongPollBotSettings(private val botSettings: TelegramBotSettings) : LongPollBotSettings {

    constructor(botUsername: String, token: String) :
            this(
                TelegramBotSettings.WithoutProxyDefaultTelegramBotSettings(botUsername, token)
            )

    constructor(botUsername: String, token: String, proxySettings: ProxySettings) :
            this(
                TelegramBotSettings.WithProxyDefaultTelegramBotSettings(botUsername, token, proxySettings)
            )

    override fun botUsername(): String = botSettings.botUsername()

    override fun token(): String = botSettings.token()

    override fun proxySettings(): ProxySettings? = botSettings.proxySettings()

}